src/frontend/sharedComponents/ProjectInviteBottomSheet/InviteBottomSheetContent/index.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import {View} from 'react-native';
+import {CommonActions} from '@react-navigation/native';
+
+import {rootNavigationRef} from '../../../AppNavigator';
+import {SessionInvite} from '../';
+import {InviteAccepted} from './InviteAccepted';
+import {InviteCanceled} from './InviteCanceled';
+import {InviteErrorOccurred} from './InviteErrorOccurred';
+import {InvitePending} from './InvitePending';
+
+export function InviteBottomSheetContent({
+  onAccept,
+  onDismiss,
+  onReject,
+  sessionInvite,
+  startConfirmationFlow,
+}: {
+  onReject: () => void;
+  onDismiss: () => void;
+  onAccept: () => void;
+  sessionInvite: SessionInvite;
+  startConfirmationFlow: () => void;
+}) {
+  const {projectName, inviteId} = sessionInvite.invite;
+
+  if (sessionInvite.status === 'pending') {
+    return (
+      <InvitePending
+        inviteId={inviteId}
+        onClose={onDismiss}
+        onReject={onReject}
+        projectName={projectName}
+        startConfirmationFlow={startConfirmationFlow}
+      />
+    );
+  }
+
+  switch (sessionInvite.removalReason) {
+    case 'accepted': {
+      return (
+        <InviteAccepted
+          projectName={projectName}
+          onGoToMap={() => {
+            if (rootNavigationRef.isReady()) {
+              onAccept();
+              rootNavigationRef.navigate('Home', {screen: 'Map'});
+            }
+          }}
+          onGoToSync={() => {
+            if (rootNavigationRef.isReady()) {
+              onAccept();
+              rootNavigationRef.dispatch(
+                CommonActions.reset({
+                  index: 1,
+                  routes: [{name: 'Home'}, {name: 'Sync'}],
+                }),
+              );
+            }
+          }}
+        />
+      );
+    }
+
+    case 'canceled': {
+      return <InviteCanceled projectName={projectName} onClose={onDismiss} />;
+    }
+
+    case 'connection error':
+    case 'internal error': {
+      return (
+        <InviteErrorOccurred projectName={projectName} onClose={onDismiss} />
+      );
+    }
+
+    // TODO: What to do when reason is "accepted other"?
+    case 'accepted other':
```

Comment:
don't think we'd run into this given how the interface is designed, but just highlighting in case there are easy suggestion to handle it more properly